### PR TITLE
Feature/user crud

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
## Summary
- Creates the CRUD for users and changes the prisma user_rol to user_role

### CODE REVIEW

File: seeders/usersSeeder.ts [hunk 1]
```diff
@@ -33,10 +33,14 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
             mode: "age",
           }),
           city: faker.address.cityName(),
+          contact_email: random(0, 1) === 1 ? true : false,
+          contact_sms: random(0, 1) === 1 ? true : false,
+          contact_wpp: random(0, 1) === 1 ? true : false,
```
Comment (ht1204) [inline seeders/usersSeeder.ts:38]: As I mentioned in an above PR:
```js
contact_wpp: random(0, 1) === 1
```
The strict comparision brings the boolean value.

File: seeders/usersSeeder.ts [hunk 2]
```diff
@@ -57,10 +61,10 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
       document_number: random(100000, 999999999999).toString(),
       date_of_birth: faker.date.birthdate({ min: 18, max: 80, mode: "age" }),
       city: faker.address.cityName(),
-      contact_email: random(0, 1) === 1,
-      contact_sms: random(0, 1) === 1,
-      contact_wpp: random(0, 1) === 1,
-      user_rol: userRoles.user,
+      contact_email: random(0, 1) === 1 ? true : false,
+      contact_sms: random(0, 1) === 1 ? true : false,
+      contact_wpp: random(0, 1) === 1 ? true : false,
```
Comment (ht1204) [inline seeders/usersSeeder.ts:66]: same here

File: seeders/usersSeeder.ts [hunk 3]
```diff
@@ -76,10 +80,10 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
       document_number: random(100000, 999999999999).toString(),
       date_of_birth: faker.date.birthdate({ min: 18, max: 80, mode: "age" }),
       city: faker.address.cityName(),
-      contact_email: random(0, 1) === 1,
-      contact_sms: random(0, 1) === 1,
-      contact_wpp: random(0, 1) === 1,
-      user_rol: userRoles.appAdmin,
+      contact_email: random(0, 1) === 1 ? true : false,
+      contact_sms: random(0, 1) === 1 ? true : false,
+      contact_wpp: random(0, 1) === 1 ? true : false,
```
Comment (ht1204) [inline seeders/usersSeeder.ts:85]: and here

File: src/api/users/users.controllers.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+    
+    res.status(200).json({ message: 'User found!', data: user });
+  } catch (error: any) {
+    res.status(500).json({ message: error.message })
+  }
+ }
+
+ // create new user
+
```
Comment (ht1204) [inline src/api/users/users.controllers.ts:44]: delete this break line, the comment should be next to the function, no spaces between

File: src/api/users/users.controllers.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    const user = await createUser(req.body);
+    res.status(201).send({message : 'User created successfully', data: user});
+  } catch (error: any) {
+    console.error(error);
+    res.status(500).json({ message: error.message })
+  }
+ }
+
+ // update user
+
```
Comment (ht1204) [inline src/api/users/users.controllers.ts:59]: same here

File: src/routes.ts [hunk 6]
```diff
@@ -1,8 +1,10 @@
 import { Application } from "express";
 
 import healthcheck from "./api/healthcheck";
+import users from "./api/users"
```
Comment (ht1204) [inline src/routes.ts:4]: semicolon here please.

File: src/api/users/users.services.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+      last_name: input.last_name,
+      document_type: input.document_type,
+      document_number: input.document_number,
+      date_of_birth: dateOfBirth,
+      city: input.city,
+      contact_email: Boolean(input.contact_email),
+      contact_sms: Boolean(input.contact_sms),
+      contact_wpp: Boolean(input.contact_wpp),
+      user_role: input.user_role
+    }
```
Comment (ht1204) [inline src/api/users/users.services.ts:45]: The input could be destructured and the could could be the following:
```js
const dateOfBirth = new Date(date_of_birth);
 return prisma.users.create({
    data: {
      email,
      password,
      name,
      last_name,
      document_type,
      document_number,
      date_of_birth: dateOfBirth,
      city
      contact_email: Boolean(contact_email),
      contact_sms: Boolean(contact_sms),
      contact_wpp: Boolean(contact_wpp),
      user_role
    }
```

File: src/api/users/users.services.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    },
+    data: {
+      email: email && { set: email },
+      password: password && { set: password },
+      name: name && { set: name },
+      last_name: last_name && { set: last_name },
+      city: city && { set: city },
+      contact_email: contact_email && { set: Boolean(contact_email) },
+      contact_sms: contact_sms && { set: Boolean(contact_sms) },
+      contact_wpp: contact_wpp && { set: Boolean(contact_wpp) },
```
Comment (ht1204) [inline src/api/users/users.services.ts:66]: Excellent

File: src/api/users/users.services.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+      last_name: true,
+      city: true,
+      user_role: true
+    }
+  });
+}
+
+
+// get a single user by the id 
+
```
Comment (ht1204) [inline src/api/users/users.services.ts:20]: delete this break line, please.

File: src/api/users/users.controllers.ts [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+ } from "./users.services";
+
+ // gets all the users from the bd
+ export const getAllUsersController = async (
+  req: Request,
+  res: Response,
+ ) => {
+  try {
+    const users = await getAllUsers()    ;
```
Comment (ht1204) [inline src/api/users/users.controllers.ts:17]: Please, remove unnecessary spaces:
```js
const users = await getAllUsers();
```

File: seeders/usersSeeder.ts [hunk 11]
```diff
@@ -33,10 +33,14 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
             mode: "age",
           }),
           city: faker.address.cityName(),
+          contact_email: random(0, 1) === 1 ? true : false,
+          contact_sms: random(0, 1) === 1 ? true : false,
+          contact_wpp: random(0, 1) === 1 ? true : false,
+          user_role: userRoles[1],
```
Comment (ht1204) [inline seeders/usersSeeder.ts:39]: Please, use the object:
```js
const userRoles = {
    user: "user",
    restaurantAdmin: "restaurantAdmin",
    appAdmin: "appAdmin",
  };
```

File: seeders/usersSeeder.ts [hunk 12]
```diff
@@ -57,10 +61,10 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
...
       date_of_birth: faker.date.birthdate({ min: 18, max: 80, mode: "age" }),
       city: faker.address.cityName(),
-      contact_email: random(0, 1) === 1,
-      contact_sms: random(0, 1) === 1,
-      contact_wpp: random(0, 1) === 1,
-      user_rol: userRoles.user,
+      contact_email: random(0, 1) === 1 ? true : false,
+      contact_sms: random(0, 1) === 1 ? true : false,
+      contact_wpp: random(0, 1) === 1 ? true : false,
+      user_role: userRoles[0],
```
Comment (ht1204) [inline seeders/usersSeeder.ts:67]: Same here, please use the object:
```js
const userRoles = {
    user: "user",
    restaurantAdmin: "restaurantAdmin",
    appAdmin: "appAdmin",
  };
```

File: src/api/users/users.controllers.ts [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+    res.status(200).send({ message: 'Users retrieved successfully', data: users });
+  } catch (error: any) {
+    res.status(500).json({message: error.message});
+  }
+ }
+
+ // get single user with id
+
```
Comment (ht1204) [inline src/api/users/users.controllers.ts:25]: Please, remove the break line

File: src/api/users/users.controllers.ts [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+    const { id } = req.params;
+    const user = await updateUser(id, req.body);
+    res.status(200).json({ message: 'User updated', data: user });
+  } catch (error: any) {
+    res.status(500).json({message: error.message});
+  }
+ }
+
+ // delete user
+
```
Comment (ht1204) [inline src/api/users/users.controllers.ts:74]: same here

**Review by ht1204 — CHANGES_REQUESTED** [review]
Please address the comments and confirm that each comment is resolved in next commit, thanks in advance.

File: seeders/usersSeeder.ts (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply seeders/usersSeeder.ts]: resolved

File: seeders/usersSeeder.ts (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply seeders/usersSeeder.ts]: resolved

File: src/api/users/users.controllers.ts (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.controllers.ts]: resolved

File: src/routes.ts (hunk 6 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/routes.ts]: resolved

File: src/api/users/users.services.ts (hunk 7 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.services.ts:45]: resolved

File: src/api/users/users.services.ts (hunk 9 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.services.ts]: resolved

File: src/api/users/users.controllers.ts (hunk 10 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.controllers.ts]: resolved

File: seeders/usersSeeder.ts (hunk 11 above)
Comment (Nicolas-Rodriguez-Ch) [reply seeders/usersSeeder.ts]: resolved

File: seeders/usersSeeder.ts (hunk 12 above)
Comment (Nicolas-Rodriguez-Ch) [reply seeders/usersSeeder.ts]: resolved

File: src/api/users/users.controllers.ts (hunk 14 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.controllers.ts]: resolved

**Comment by Nicolas-Rodriguez-Ch** [comment]
code modified to take into account the feedback

File: src/api/users/users.controllers.ts (hunk 5 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.controllers.ts]: resolved

File: src/api/users/users.controllers.ts (hunk 13 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.controllers.ts]: resolved

File: seeders/usersSeeder.ts [hunk 15]
```diff
@@ -76,10 +76,10 @@ const seedUsers = async (prisma: PrismaClient): Promise<void> => {
       document_number: random(100000, 999999999999).toString(),
       date_of_birth: faker.date.birthdate({ min: 18, max: 80, mode: "age" }),
       city: faker.address.cityName(),
-      contact_email: random(0, 1) === 1,
-      contact_sms: random(0, 1) === 1,
-      contact_wpp: random(0, 1) === 1,
-      user_rol: userRoles.appAdmin,
+      contact_email: random(0, 1) === 1 ? true : false,
+      contact_sms: random(0, 1) === 1 ? true : false,
+      contact_wpp: random(0, 1) === 1 ? true : false,
```
Comment (ht1204) [inline seeders/usersSeeder.ts]: But why using ternary operator? The strong comparison would do the job returning the boolean value of the operation.  As I mentioned above, consider `random(0,1) === 1`.

File: seeders/usersSeeder.ts (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply seeders/usersSeeder.ts]: resolved

**Review by ht1204 — APPROVED** [review]
Good job, move forward!

**Pull request merged**